RIOT-OS/RIOT
# tests/pkg_micro-ecc: Cleanup and AVR fixes

### Contribution description

The test in `tests/pkg_micro-ecc` performed some huge allocation on the stack, including variable length arrays. Both should be avoided in general. But more specifically this causes a stack overflow on ATmega boards. This PRs moves the allocations from the stack to the data / bss segment to solve the stack overflow. In addition, the verbosity of the output is increased. (E.g. when running `make term` the terminal will only print the line once the `\n` is received, which is at the end of the test. The test takes about 10 minutes on an ATmega clocked at 8MHz, or about 5 minutes on an ATmega-Arduino clocked at 16 MHz. No user will wait for that time for output to appear on the terminal.)

### Testing procedure

Run:
```
make BOARD=<foo> flash test -C tests/pkg_micro-ecc
```
It should still work with a non ATmega board and should now succeed with ATmega boards.

### Issues/PRs references

Tick one item in https://github.com/RIOT-OS/RIOT/issues/12651
